src/app/AppContainer.js

```diff
@@ -47,8 +53,13 @@ class AppContainer extends React.Component<PropsType> {
 			type: 'draft',
 		};
 
-		this.props.setupEditor( post );
+		this.state = {
+			title: props.title,
```

Comment:
This brings up two questions in my head.

The first one is: As I understand, this is to keep in line with the rest of the architecture, or is there a technical reason why this is necessary before merging the PR?

The second question is related to the first one: Is this urgent or can I work on it incrementally?  I'd like if possible to work on more atomic PRs to avoid monolithic ones which can become very complex very fast.

Reply:
I think the number of lines would probably decrease after using the store because we'll be getting rid of propagating title change through props and we'll remove the state, and considering that it'll require same tests it doesn't look like an incremental change to me. The technical reason could be that currently the store doesn't know about the title at all, so I feel like we'll need to address this sooner or later but I don't want to block or get in the way of anything. Maybe other reviewers have other opinions on this.

Reply:
> I think the number of lines would probably decrease after using the store because we'll be getting rid of propagating title change through props and we'll remove the state, and considering that it'll require same tests it doesn't look like an incremental change to me. 

Working incrementally is 100% about simplifying the steps.

My reasoning is:

- The PR is working right now, all the way to WPiOS.
- I can deliver something today and work to make it better.  I commit to this!
- We can close 3 PRs and avoid having to keep resolving conflicts meanwhile.

> The technical reason could be that currently the store doesn't know about the title at all, so I feel like we'll need to address this sooner or later but I don't want to block or get in the way of anything. Maybe other reviewers have other opinions on this.

Right now, Aztec is being able to query the title as it is.

Again, I'm definitely not advocating to do things the wrong way... just proposing that we split the steps.  It's only a difference in methodology.

From what I can tell, we should also get the feature sooner this way.

Reply:
We can open a separate issue and move fwd on this I think. And It'd be good to have opinion of @hypest also. I'll keep on reviewing this in parallel I couldn't finish it yet.

Reply:
👋. I think it's OK to work by splitting to separate steps, even if a step will actually revise code written from a previous step. Reason is that it can help scope down the change (note, not the same as minimizing lines-of-code change) and review it easier/faster.

What I would recommend though is to at least add some inline comments in the places that we're pretty sure we'll touch in a follow up PR. For example, a suitable comment in this case is that we'd like to move the state to the Redux Store (`data`) so, this local state is only a intermediate step. Hope this makes sense!

Reply:
I still don't have the knowledge to know what'll need to be changed or where.  Sorry about that!

But since we'll need to wait a bit on the Android part I'll try branching off this PR and pushing forward the changes in parallel.

Reply:
> I still don't have the knowledge to know what'll need to be changed or where. Sorry about that!

No prob! Let's just handle the cases already commented about, cool?

@Tug , can you have a look at the Gutenberg-web codebase to figure out where is the post title stored? Is it in the Redux store or elsewhere? @diegoreymendez let's add a comment about this when Tug gets insight on this one, cool?

> But since we'll need to wait a bit on the Android part I'll try branching off this PR and pushing forward the changes in parallel.
>
> Sounds good?

That's fine, sure. I'm mainly trying to care about the points already brought up during this review so we don't lose sight of them. We can always handle/fix/update things in follow up PRs.

Reply:
the title is part of the post:
```
const post = {
	id: 1,
	title: {
		raw: props.title,
	},
	content: {
		raw: props.initialHtml,
	},
	status: 'draft',
};
```
We don't need to use the component state for that, it's in the global store.

We can then access/modify it using `getEditedPostAttribute( 'title' )` and `editPost( { title } )` and it's added to the list of edits!
See the exemple here: https://github.com/WordPress/gutenberg/tree/master/packages/editor/src/components/post-title/index.js

Reply:
Thanks @Tug.  While I didn't tackle this in this PR, I'm now opening a new one to take care of it.

This was very valuable and insightful.